build_scripts/colab-tests/validate_colab_dependencies.py

```diff
@@ -0,0 +1,85 @@
+import pathlib
+from poetry.core.semver.version import Version
+
+from collections import defaultdict
+from poetry.factory import Factory
+import re
+
+
+# If you add any new entries to this list, please make sure you test your changes in Colab manually first!
+IGNORED_VIOLATIONS = {
+    "Version check failed for 'click'. Colab has 7.1.2, but 'flask' requires >=8.0",
+    "Version check failed for 'cloudpickle'. Colab has 1.3.0, but 'layer' requires >=2.0.0",
+    "Version check failed for 'grpcio'. Colab has 1.46.3, but 'grpcio-tools' requires >=1.47.0",
+    "Version check failed for 'humanize'. Colab has 0.5.1, but 'layer' requires >=3.11.0",
+    "Version check failed for 'itsdangerous'. Colab has 1.1.0, but 'flask' requires >=2.0",
+    "Version check failed for 'protobuf'. Colab has 3.17.3, but 'layer-api' requires >=3.19.4",
+    "Version check failed for 'pyarrow'. Colab has 6.0.1, but 'layer' requires 7.0.0",
+    "Version check failed for 'urllib3'. Colab has 1.24.3, but 'botocore' requires >=1.25.4,<1.27",
+}
```

Comment:
what is the plan for those?

Reply:
No plan at the moment. I don't think we can eliminate them all. 

I see this test as linter - if we make a conscientious decision to ignore an error that is fine, but at least we will find out about these.

These are not (currently) fatal - we update them in Colab and things still work.